attribute-service-impl/src/main/java/org/hypertrace/core/attribute/service/delegate/AttributeUpdaterImpl.java

```diff
@@ -80,14 +79,20 @@ private void validate(final UpdateMetadataRequest request) {
   private FilterTypeExpression buildFilter(
       final UpdateMetadataRequest request, final String tenantId) {
     final FilterTypeExpression idFilter = filterBuilder.buildIdFilter(request.getAttributeId());
+
+    // Not filtering by the tenant hierarchy to update only the custom attributes
     final FilterTypeExpression tenantIdFilter = filterBuilder.buildTenantIdFilter(tenantId);
+
     return and(idFilter, tenantIdFilter);
   }
 
   private List<SubDocumentUpdate> buildUpdates(final UpdateMetadataRequest request) {
-    return request.getUpdatesList().stream()
-        .map(updateBuilder::buildUpdate)
-        .flatMap(Optional::stream)
-        .collect(toUnmodifiableList());
+    try {
+      return request.getUpdatesList().stream()
+          .map(updateBuilder::buildUpdate)
+          .collect(toUnmodifiableList());
+    } catch (final UnsupportedOperationException e) {
```

Comment:
ah - I mentioned this elsewhere, but this works too. That said, why not just throw the status directly? Would be cleaner and no implicit contract.